### PR TITLE
Create UserProfile model and implement optimistic locking

### PR DESCRIPTION
**Pull request by sslaws**
Create UserProfile (simple)
Implement optimistic locking

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_204)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_204)

File: Apps/Database/src/Context/BaseDbContext.cs [hunk 1]
```diff
@@ -36,6 +37,40 @@ public BaseDbContext(DbContextOptions options)
...
+
+        /// <summary>
+        /// Gets the initial seed date for loading data.
+        /// The value returned is the first date of the Health Gateway Project.
+        /// </summary>
+        protected virtual DateTime DefaultSeedDate
+        {
+            get
+            {
+                return Convert.ToDateTime("5/1/2019", CultureInfo.InvariantCulture);
```
Comment (FuriousLlama) [inline Apps/Database/src/Context/BaseDbContext.cs]: May 1st?

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_204)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_204)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_204)

**Pull request merged**